enarx-keep-sev-shim/src/gdt.rs

```diff
@@ -59,93 +51,59 @@ pub fn init() {
         stack_start + num_pages * Page::size()
     }
 
-    unsafe {
-        TSS = Some({
-            let mut tss = TaskStateSegment::new();
-
-            static mut STACKS: [[Page; STACK_NUM_PAGES]; 8] =
-                [[Page::zeroed(); STACK_NUM_PAGES]; 8];
-
-            // The stack for RING 0
-            tss.privilege_stack_table[0] =
-                stack_end(VirtAddr::from_ptr(&STACKS[0]), STACK_NUM_PAGES);
+    TSS.privilege_stack_table[0] = level_0_stack;
 
-            // The stacks for the exceptions and interrupts
-            for i in 0..7 {
-                tss.interrupt_stack_table[i] =
-                    stack_end(VirtAddr::from_ptr(&STACKS[i + 1]), STACK_NUM_PAGES);
-            }
-            tss
-        });
+    // The stacks for the exceptions and interrupts
+    for i in 0..7 {
```

Comment:
It might be cleaner to use zip on iterators here.

Reply:
ok

Reply:
Because zip() ends, when one if the iterators is short on elements, I added:
```rust
debug_assert_eq!(STACKS.len(), TSS.interrupt_stack_table.len());
```